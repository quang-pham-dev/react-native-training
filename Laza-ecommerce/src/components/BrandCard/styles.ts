import { StyleSheet } from 'react-native';

// Themes
import Colors from 'themes/Colors';
import Metrics from 'themes/Metrics';

const styles = StyleSheet.create({
  brandCardContainer: {
    justifyContent: 'center',
  },
  brandCardWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    borderRadius: 10,
    height: 50,
    backgroundColor: Colors.secondaryBackground,
    marginRight: Metrics.margin.sm,
    paddingRight: Metrics.padding.sm,
  },
  brandLogoWrapper: {
    width: 40,
    height: 40,
    padding: Metrics.padding.xs,
    borderRadius: 10,
    marginLeft: Metrics.margin.xsm,
    marginRight: Metrics.margin.sm,
    marginVertical: Metrics.margin.xsm,
    backgroundColor: Colors.primaryBackground,
  },
  brandLogo: {
    width: '100%',
    height: '100%',
    resizeMode: 'contain',
  },
});

export default styles;
