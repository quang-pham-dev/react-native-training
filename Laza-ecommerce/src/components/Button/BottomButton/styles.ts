import { StyleSheet } from 'react-native';
// Styles
import { Metrics, Colors, Fonts } from '@styles/themes/index';

const styles = StyleSheet.create({
  button: {
    paddingVertical: Metrics.padding.md,
    paddingHorizontal: Metrics.padding.xl,
    alignItems: 'center',
  },
  title: {
    color: Colors.white,
    textTransform: 'none',
    fontFamily: Fonts.fontFamily.Inter_600SemiBold,
    fontSize: Fonts.size.sm,
    lineHeight: Fonts.lineHeight.lg,
  },
});

export default styles;
