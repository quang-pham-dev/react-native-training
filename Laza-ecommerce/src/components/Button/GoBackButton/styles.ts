// Styles
import { Metrics } from '@styles/themes/index';
import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  button: {
    marginTop: Metrics.margin.customXxl,
    marginLeft: Metrics.margin.lg,
    alignItems: 'center',
  },
  icon: {
    width: Metrics.icons.semiLarge,
    height: Metrics.icons.semiLarge,
  },
});

export default styles;
