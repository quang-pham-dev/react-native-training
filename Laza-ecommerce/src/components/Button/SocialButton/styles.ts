import { StyleSheet } from 'react-native';
// Styles
import { Metrics, Colors, Fonts } from '@styles/themes/index';

const styles = StyleSheet.create({
  button: {
    paddingVertical: Metrics.padding.md,
    paddingHorizontal: Metrics.padding.xl,
    alignItems: 'center',
    borderColor: Colors.transparent,
    borderWidth: 1,
    backgroundColor: Colors.white,
  },
  title: {
    color: Colors.white,
    textTransform: 'none',
    fontFamily: Fonts.fontFamily.Inter_600SemiBold,
    fontSize: Fonts.size.sm,
    lineHeight: Metrics.height.custom,
  },
  icon: {
    width: Metrics.icons.custom,
    height: Metrics.icons.custom,
  },
});

export default styles;
