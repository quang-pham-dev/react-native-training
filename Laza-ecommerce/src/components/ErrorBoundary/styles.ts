import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },

  subContainer: {
    marginHorizontal: 30,
    alignItems: 'center'
  },

  icon: {
    marginBottom: 18
  },

  bigBoldText: {
    fontSize: 18,
    color: 'black',
    letterSpacing: 0.8,
    fontWeight: 'bold'
  }
});

export default styles;
