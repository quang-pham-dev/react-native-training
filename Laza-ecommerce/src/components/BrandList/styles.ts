import { StyleSheet } from 'react-native';
import { Metrics } from 'styles/themes';
const styles = StyleSheet.create({
  container: {},
  brandTitle: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingBottom: Metrics.padding.xm,
  },
});

export default styles;
