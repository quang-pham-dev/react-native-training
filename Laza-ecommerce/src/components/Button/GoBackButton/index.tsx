import React, { memo } from 'react';
import { Image, TouchableOpacity } from 'react-native';
// Types
import { GoBackButtonProps } from 'types/Button';
// Styles
import styles from './styles';

const GoBackButton = ({
  icon,
  disabled,
  buttonStyles = [],
  onPress = () => {},
}: GoBackButtonProps) => {
  return (
    <TouchableOpacity
      style={[styles.button, buttonStyles]}
      onPress={onPress}
      disabled={disabled}
      activeOpacity={0.8}
    >
      {icon && <Image style={styles.icon} source={icon} />}
    </TouchableOpacity>
  );
};

export default memo(GoBackButton);
