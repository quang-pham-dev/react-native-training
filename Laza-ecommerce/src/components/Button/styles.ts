import { StyleSheet } from 'react-native';
import { Colors } from 'styles/themes';

const styles = StyleSheet.create({
  icon: {},
  text: {
    color: Colors.white,
  },
});
export default styles;
