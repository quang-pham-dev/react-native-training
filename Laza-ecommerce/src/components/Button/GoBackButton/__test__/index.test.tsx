import React from 'react';
import GoBackButton from 'components/Button/GoBackButton';
import renderer from 'react-test-renderer';
import { IMAGES } from 'styles/themes';

describe('Social Button Component', () => {
  test('should render  correctly', () => {
    const tree = renderer
      .create(<GoBackButton icon={IMAGES.iconBack} onPress={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
