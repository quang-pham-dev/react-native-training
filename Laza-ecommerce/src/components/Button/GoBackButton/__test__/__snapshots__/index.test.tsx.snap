// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social Button Component should render  correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "marginLeft": 20,
      "marginTop": 45,
      "opacity": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 45,
        "width": 45,
      }
    }
  />
</View>
`;
